presto-resource-group-managers/src/main/java/io/prestosql/plugin/resourcegroups/ResourceGroupSpec.java

```diff
@@ -71,9 +70,7 @@ public ResourceGroupSpec(
         checkArgument(maxQueued >= 0, "maxQueued is negative");
         this.maxQueued = maxQueued;
         this.softConcurrencyLimit = softConcurrencyLimit;
-
-        checkArgument(hardConcurrencyLimit.isPresent() || maxRunning.isPresent(), "Missing required property: hardConcurrencyLimit");
-        this.hardConcurrencyLimit = hardConcurrencyLimit.orElseGet(maxRunning::get);
+        this.hardConcurrencyLimit = hardConcurrencyLimit;
         checkArgument(this.hardConcurrencyLimit >= 0, "hardConcurrencyLimit is negative");
```

Comment:
I'm thinking this should be `> 0`. If you don't specify `hardConcurrencyLimit`, it will be zero by default, which might cause all queries to get queued. We probably missed fixing this check originally, because no one would intentionally specify zero.

Reply:
Right, it should be `> 0`. 

> will be zero by default

i don't understand this. `hardConcurrencyLimit` is a required field. Does Jackson fill in required `int` fields with 0?

Reply:
Yes, since `FAIL_ON_MISSING_CREATOR_PROPERTIES` is false by default. However, I just learned that there is a `required` property for `@JsonProperty` on `@JsonCreator` which can be used to fail in this case. Simply using a primitive is not the same as making it required.

Regardless of whether we change that or not, I don't think we should allow users to set the value to zero, as queuing forever is not a useful behavior.

Reply:
> However, I just learned that there is a required property for @JsonProperty 

I even such code change in this PR for a while. But we don't use `required=true` probably anywhere else, so i reverted.

should we enable `FAIL_ON_MISSING_CREATOR_PROPERTIES` for reading the RG config file?

I agree that it makes sense to require positive value. However, we cannot enforce this in the constructor of the spec class, as it's instantiated by DB RG manager.
What about adding `> 0` check for this property in the File RG loading code and retaining `>= 0` in this class?

Reply:
> should we enable FAIL_ON_MISSING_CREATOR_PROPERTIES for reading the RG config file?

The Jackson docs say

>> Feature is disabled by default, so that no exception is thrown for missing creator property values, unless they are explicitly marked as `required`.

I believe that means we would need to both enable that flag and mark them as `required=true`. Though Jackson behavior is often counterintuitive so this would require testing to make sure it works as expected.

For config files, which is a rather different way of using Jackson than our normal usage, this seems like a good idea. Getting `0` as a default is unexpected, both for us and for users.

> What about adding `> 0` check for this property in the File RG loading code and retaining `>= 0` in this class?

Sounds good to me